### DISCUSSION TITLE
CORS Error(No 'Access-Control-Allow-Origin' header is present on the requested resource)-Laravel Deployment

### DISCUSSION THREAD

**Discussion by SWS-5007** [vercel/community#1213] [Help]
### Summary

Hi, I am deploying the Laravel Project on Vercel.
When Frontend send request to this Laravel Backend, I am getting the Cors Error like below:

> Access to XMLHttpRequest at 'https://businesscard-backend.vercel.app/api/getTemplates' from origin 'https://businesscard-frontend.netlify.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

Here is my `vercel.json` file of Laravel Project.
```
{
    "version": 2,
    "functions": {
        "api/index.php": {
            "runtime": "vercel-php@0.5.2"
        }
    },
    "routes": [{
        "src": "/(.*)",
        "dest": "/api/index.php"
    }],
    "headers": [{
        "source": "/api/(.*)",
        "headers": [{
                "key": "Access-Control-Allow-Credentials",
                "value": "true"
            },
            {
                "key": "Access-Control-Allow-Origin",
                "value": "https://businesscard-frontend.netlify.app"
            },
            {
                "key": "Access-Control-Allow-Methods",
                "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT"
            },
            {
                "key": "Access-Control-Allow-Headers",
                "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version"
            }
        ]
    }],
    "env": {
        "APP_NAME": "BusinessCard-Backend",
        "APP_ENV": "production",
        "APP_DEBUG": "false",
        "APP_CONFIG_CACHE": "/tmp/config.php",
        "APP_EVENTS_CACHE": "/tmp/events.php",
        "APP_PACKAGES_CACHE": "/tmp/packages.php",
        "APP_ROUTES_CACHE": "/tmp/routes.php",
        "APP_SERVICES_CACHE": "/tmp/services.php",
        "VIEW_COMPILED_PATH": "/tmp",
        "CACHE_DRIVER": "array",
        "LOG_CHANNEL": "stderr",
        "SESSION_DRIVER": "cookie",
        "VIEW_COMPILED_PATH": "/tmp/views",
        "SSR_TEMP_PATH": "/tmp/ssr",
        "NODE_PATH": "node"
    }
}
```
What should I do to fix this CORS Error?
Please help me!

**Comment by TopStar-57** [comment]
I am getting same error.
When you find the solution, please let me know too.
Thanks

**Comment by amyegan** [comment]
I'm getting a 404 from https://businesscard-backend.vercel.app/api/getTemplates. Guessing that's because of a missing token or query value when I visit the URL directly. Could it be that your frontend is running into a similar problem and the error is actually because the request is also missing the right auth token (or other expected value) and getting an unexpected response?

If you are able to share your repo or a minimal reproduction, I'm happy to help with debugging. Otherwise, [this post on Stack Overflow](https://stackoverflow.com/questions/43871637/no-access-control-allow-origin-header-is-present-on-the-requested-resource-whe) has a lot of suggestions for handling CORS errors.

**Reply by SWS-5007** [reply]
Hi, @amyegan 
Thanks so much for your answer.
I shared the repo with you as you said.
Please check this:
[https://github.com/SWS-5007/BusinessCard-Backend/tree/vercel_deploy](https://github.com/SWS-5007/BusinessCard-Backend/tree/vercel_deploy
)

And check this too.
[https://business-card-frontend.vercel.app/](https://business-card-frontend.vercel.app/)
There is my main CORS error now.
> Access to XMLHttpRequest at 'https://businesscard-backend.vercel.app/api/getInfos' from origin 'https://business-card-frontend.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: It does not have HTTP ok status.

How can I solve this problem?
Please help me.

**Comment by SMNahuel** [comment]
I same error please help

**Reply by SMNahuel** [reply]
I solved this error

**Reply by jaroslavez** [reply]
How did you solve it?

**Reply by SMNahuel** [reply]
You have a module csrf in you app, in app, Http, Kernel.php, I'm not working always in laravel this problem you can see my proyect in this link https://github.com/Camilaescalant/final_web1-2/tree/main

**Comment by vickyleon** [comment]
same error

**Comment by imadovetch** [comment]
did you solve the error man ?? i need the solution

**Reply by SMNahuel** [reply]
I solved deleted crsf in module http

**Reply by imadovetch** [reply]
do you mean this middlware 
public function handle($request, Closure $next)
{
    // Comment out the CSRF token check
    // $this->validateCsrf($request);

    return $next($request);
}

do you have a repoo i can fork it and try hosting it

**Reply by SMNahuel** [reply]
https://github.com/Camilaescalant/final_web1-2 is this

**Reply by imadovetch** [reply]
ok i clonned the repo can you give me the vercel.json config

**Reply by adnanerlansyah403** [reply]
Do you have already fixed this issue? Or is there any config of vercel.json that was found?
